cunumeric/array.py

```diff
@@ -2296,6 +2446,7 @@ def nancumprod(
     # currently offset option is implemented only for the case of number of
     # axes=2. This restriction can be lifted in the future if there is a
     # use case of having arbitrary number of offsets
+    @add_boilerplate()
```

Comment:
was there a case where this was really necessary? `_diag_helper` is an internal method that the user is not supposed to call directly and I believe all entry points that invoke this are already wrapped with `add_boilerplate`.

Reply:
d75bbbeb85a333bf69b98a5a82b50c58f822d86b